main.py

```diff
@@ -415,57 +422,18 @@ def compare_semi_crash(expected_log_path, actual_log_path, bug_pattern_data):
 
     logs_to_match = []
 
+    # get logs based on defined pattern/grammer from expected log file
     for key in bug_pattern_data:
         pattern = bug_pattern_data[key]
-        content = re.search(pattern, expected_content).group(1)
+        content = re.search(pattern, expected_content)
+        if not content:
+            continue   
+        content = content.group(1) 
+        if key == "file_pattern":
+            content = os.path.basename(content)
         logs_to_match.append(content)
-    return all(string in actual_content for string in logs_to_match)
-
-
-def compare_false_positive_log(expected_log_path, actual_log_path,  bug_pattern_data):
-    with open(expected_log_path, "r") as file:
-        expected_content = file.read()
-
-    with open(actual_log_path, "r") as file:
-        actual_content = file.read()
-    
-    file_pattern = bug_pattern_data["file_pattern"]
-    error_pattern = bug_pattern_data["error_pattern"]
-    source_pattern = bug_pattern_data["source_pattern"]
-    found_pattern = bug_pattern_data["found_pattern"]
-    required_pattern = bug_pattern_data["required_pattern"]
-
-    java_file = re.search(file_pattern, expected_content).group(1)
-    error_message = re.search(error_pattern, expected_content).group(1)
-    code_triggered_bug = re.search(source_pattern, expected_content).group(1)
-    found_type = re.search(found_pattern, expected_content).group(1)
-    required_type = re.search(required_pattern, expected_content).group(1)
-
-    return java_file in actual_content and error_message in actual_content and code_triggered_bug in actual_content and found_type in actual_content and required_type in actual_content
-
-
-def compare_error_log(expected_log_path, actual_log_path, bug_pattern_data):
-    '''
-    Compare the error log of the minimized program with the expected error log
-    '''
-    with open(expected_log_path, "r") as file:
-        expected_content = file.read()
-
-    with open(actual_log_path, "r") as file:
-        actual_content = file.read()
-    
-    file_pattern = bug_pattern_data["file_pattern"]
-    error_pattern = bug_pattern_data["error_pattern"]
-    source_pattern = bug_pattern_data["source_pattern"]
-    reason_pattern = bug_pattern_data["reason_pattern"]
-
-    error_file = re.search(file_pattern, expected_content).group(1)
-    error_message = re.search(error_pattern, expected_content).group(1)
-    error_source = re.search(source_pattern, expected_content).group(1)
-    error_reason = re.search(reason_pattern, expected_content).group(1)
-
-    return error_file in actual_content and error_message in actual_content and error_source in actual_content and error_reason in actual_content
 
+    return all(string in actual_content for string in logs_to_match)
```

Comment:
> I can run the pattern extraction steps on minimized program build log file also and check if both extracted data are same. In that way we can say both log files are of same issue.

I think it's important that we do this. Ideally, the algorithm that I expect is:
* use the patterns to extract matching content from the expected log
* use the same patterns to extract content from the actual log
* return true iff there is an exact match between the two